src/verify/rsa.rs

```diff
@@ -0,0 +1,104 @@
+use der::Reader;
+use paste::paste;
+use pki_types::{AlgorithmIdentifier, InvalidSignature, SignatureVerificationAlgorithm};
+use rsa::{pkcs1v15, pss, BigUint, RsaPublicKey};
+use sha2::{Sha256, Sha384, Sha512};
+use signature::Verifier;
+use webpki::alg_id;
+
+fn decode_spki_spk(spki_spk: &[u8]) -> Result<RsaPublicKey, InvalidSignature> {
+    // public_key: unfortunately this is not a whole SPKI, but just the key
+    // material. decode the two integers manually.
+    let mut reader = der::SliceReader::new(spki_spk).map_err(|_| InvalidSignature)?;
+    let ne: [der::asn1::UintRef; 2] = reader.decode().map_err(|_| InvalidSignature)?;
+
+    RsaPublicKey::new(
+        BigUint::from_bytes_be(ne[0].as_bytes()),
+        BigUint::from_bytes_be(ne[1].as_bytes()),
+    )
+    .map_err(|_| InvalidSignature)
+}
```

Comment:
This format is actually PKCS#1.

You can use [`RsaPublicKey::from_pkcs1_der`](https://docs.rs/rsa/latest/rsa/#pkcs1-rsa-key-encoding) to decode it

Reply:
awesome, didn't know it exists